circularBuffer.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include "circularBuffer.hpp"
+
+template <class T> T      circularBuffer<T>::buffer[BufferSize] = {};
+template <class T> size_t circularBuffer<T>::head =                0;
+template <class T> size_t circularBuffer<T>::tail =                0;
+template <class T> bool   circularBuffer<T>::full =            false;
+
+template <class T> void circularBuffer<T>::put_overwrite(T item)
```

Comment:
Why both here and in the header?

Reply:
Sorry its a bug. circularbuffer.cpp is not needed.